submissions/eoneoff/port-sniffer/sniffer.py

```diff
@@ -0,0 +1,153 @@
+            should be definded as two numbers with a dash between them. If not set
+            or set incorrectly the default range of (1-65535) is scanned
+
+                python sniffer.py --host wwww.google.com --ports 1024-2018
+
+--help      parameter to see this reference
+
+To import port scanner use
+
+    import sniffer.py
+
+after import you can perform scan by calling sniffer.scan()
+
+scan(host[, portString][, module][, silent])
+
+For  detailed disambiguation of scan function use help(scan) after importing
+sniffer or call sniffer.py --help with 'method' parameter
+"""
+
+
+import socket
+import sys
+import re
+
+def _twister():
+    symbols = ['|', '/', '-', '\\']
+    current = 0
+    while True:
+        yield symbols[current]
+        current = 0 if current == 3 else current + 1
+
+def _is_ip_address(host):
+    return re.search('^((25[0-5]|2[0-4]\d|[01]?\d?\d)\.){3}(25[0-5]|2[0-4]\d|[01]?\d?\d)$', host)
+
+def _parse_host(host):
+    if not _is_ip_address(host):
+        try:
+            return socket.gethostbyname(host)
+        except socket.gaierror as err:
+            err.args = "Hostname could not be resolved"
+            raise err
+    else: return host
+
+def _parse_ports(ports):
+    if not re.search('^\d{1,4}-\d{1,4}$', ports):
+        raise ValueError
+    limits = [int(i) for i in ports.split('-')]
+    if limits[0] >= limits[1] or any([i not in range(1, 65536) for i in limits]):
+        raise ValueError
+    else:
+        return range(limits[0], limits[1]+1)
+
+
+def _parse_arguments(args):
+    host = ""
+    ports = range(1, 65536)
+    if '--host' not in args or args.index('--host') == len(args)- 1:
+        raise ValueError("You must specify a host to scan ports")
+    else:
+        host = _parse_host(args[args.index('--host') + 1])
+    if '--ports' in args:
+        index = args.index('--ports') + 1
+        if index == len(args):
+            print("Invalid ports specified, setting to default (1-65535)")
+        else:
+            try:
+                ports = _parse_ports(args[index])
+            except ValueError:
+                print("Invalid ports specified, setting to default (1-65535)")
+    return(host, ports)
+
+def _scan(host, ports, module=False, silent=False):
+    socket.setdefaulttimeout(0.3)
+    open_ports = []
+    twr = _twister()
+    print('scanning', end=' ', flush=True)
+    for port in ports:
+        with socket.socket() as skt:
+            if skt.connect_ex((host, port)) == 0:
+                open_ports.append(port)
+                print('.', end=' ', flush=True)
+            else:
+                print(f'{next(twr)}\b', end='', flush=True)
+    if not silent or not module:
+        result = ''
+        if open_ports:
+            result = "Port{} {} {}".format('s' if len(open_ports) == 1 \
+                else '', ' '.join((str(p) for p in open_ports)), \
+                    "are" if len(open_ports) > 1 else "is")

```

Comment:
better to move to f-strings
also could be convenient to move all input-generation to separate method and perform it there